examples/schema-with-header-param.ts

```diff
@@ -1,13 +1,16 @@
 /* eslint-disable */
 import type { GetRequest, PostRequest, PutRequest, PatchRequest, OptionsRequest, DeleteRequest } from 'openapi-tsrf-runtime'
-import { toQuery, toFormData } from 'openapi-tsrf-runtime'
+import { toQuery, toFormData, toHeaders } from 'openapi-tsrf-runtime'
 export type TransferResponse = { }
 export abstract class RequestFactory {
-  static getAccountAccountIdTransfers({ accountId, startDate, endDate, limit, cursor, }: { accountId: string, startDate?: string, endDate?: string, limit?: number, cursor?: string, }): GetRequest<{ message: string, nextCursor?: string, data: Array<TransferResponse>, }> {
+  static getAccountAccountIdTransfers({ acceptVersion, accountId, startDate, endDate, limit, cursor, }: { acceptVersion: '1.0'
+    | '1.1', accountId: string, startDate?: string, endDate?: string, limit?: number, cursor?: string, }): GetRequest<{ message: string, nextCursor?: string, data: Array<TransferResponse>, }> {
     const query = toQuery({ startDate, endDate, limit, cursor })
+    const headers = toHeaders({ 'Accept-Version': acceptVersion })
```

Comment:
I think it's worth adding functionality to only include specific headers in the request factories as I think _most_ of the time, headers will not be something we want to have to pass in with each request.  In this example the `acceptVersion` param is something you'd specify in the proxy class rather than specify on each request. 

Perhaps we could add something like:
```shell
// Include all headers in request factories
openapi-tsrf generate ... --include-headers *

// Include specific headers in request factories
openapi-tsrf generate ... --include-headers header1 header2 header3
```

It'd also be worth including a specific example of this behaviour via a new spec in the `examples` dir and adding a matching test in `tests/index.spec.ts`

Reply:
[commander docs](https://www.npmjs.com/package/commander#variadic-option) for variadic options

Reply:
Yeah that makes sense. I'll add that in.

Reply:
I think headers come in two flavours. Ones you supply with every request and ones that are defined by the spec per operation.

In the case of the one defined by the spec, I don't know if you want to have the option to disable it.

Take this spec, for example:

```
@@ -0,0 +1,1227 @@
{
  "openapi": "3.0.0",
  "paths": {
    "/receipts": {
      "post": {
        "operationId": "receipts",
        "parameters": [
          {
            "name": "some-value",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string",
            }
          },
          {
            "name": "X-Platform",
            "in": "header",
            "required": true,
            "schema": {
              "type": "string",
            }
          }
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "required": [
                  "app_user_id",
                ],
                "properties": {
                  "app_user_id": {
                    "type": "string",
                  },
                }
              }
            }
          }
        },
        "responses": {
          "200": {
            "content": {
              "application/json": { }
            }
          }
        },
        "deprecated": false
      }
    }
  }
}
```

The spec advertises that you can only call the endpoint if you supply the required header. So, I think roughly the call would be

```
  const response = await API.postReceipts({ someValue: 'hi',  xPlatform: 'pleb os' })
```

Reply:
It's not a choice of whether or not to include the header - it's whether we force the consumer to provide it at the call site (as per your example) or whether we implicitly supply it when constructing the request in our api proxy factory. 

We can't determine which option is better from the spec alone, so we should make it configurable so the user can decide.

Reply:
As discussed, I've added a `--include-headers` flag to the cli, allowing you to specify `*` or a whitelist of header names like `--include-headers header1 header 2`

Reply:
default behaviour is to not include any headers